shakytweaks/src/main/java/com/mindsea/shakytweaks/ShakyTweaks.kt

```diff
@@ -61,6 +67,21 @@ object ShakyTweaks {
         sensorManager.registerListener(shakeDetector, accelerometer, SensorManager.SENSOR_DELAY_UI)
     }
 
+    fun init(activity: Activity, lifecycle: Lifecycle) {
```

Comment:
I am still worried about having activity context here. Even more because we don't actually `need` it here.

I'd rather change it to `Context`. User of this lib can still pass `this` when inside an Activity.
And for an additional level of security, we should access `applicationContext`, like this:
```
   fun init(context: Context, lifecycle: Lifecycle) {
     val applicationContext = context.applicationContext
     
     // only use applicationContext from now on
   
   }
```

Reply:
@gianei We need here because there is an action called:
```
shakeDetector.setListener {
       activity.startActivity(createTweaksActivityIntent(activity))
}
```
To call for another activity.

Reply:
@gianei I think we also can update the `createTweaksActivityIntent`, because if we have the Activity as a Context, it is not necessary we pass the `Intent.FLAG_ACTIVITY_NEW_TASK` as argument:

Take a look in this part:
```
internal fun createTweaksActivityIntent(context: Context) =
    Intent(context, TweaksActivity::class.java).apply {
        flags = Intent.FLAG_ACTIVITY_NEW_TASK
    }
```

Reply:
But, the Intent is using `Intent.FLAG_ACTIVITY_NEW_TASK` exactly so that we can start this activity from `applicationContext`...
Why shouldn't we use this approach?

Reply:
Yes, but in my opinion, we don't need more of it. 
Think about that now we have an `Activity` reference. We can launch a new `Activity` from another. 
I don't know if I'm clear enough. 
We can talk more about it if you want. 
🤗️